Workiva/over_react_codemod
# CPLAT-13196 Add Flag For Core Check

## Motivation
  
To prepare for the 2.9 corecheck enforcing that the repo isn't already transitioned, we needed to update the codemod to allow for a dry run and short circuit if no new boilerplate is found.

## Changes
  
- Add `LatestBoilerplateVisitor`, which looks for the new factory syntax
  - Add tests
  - Add `check-for-transitioning` flag
  - When the flag is set:
    -  Use visitor to check all dart files for new boilerplate
    - If there is new boilerplate, add the `--fail-on-changes` flag
    - Skip the pubspec update check
    - Run the codemod with the new flag
#### Release Notes
  

## Review
_[See CONTRIBUTING.md][contributing-review-types] for more details on review types (+1 / QA +1 / +10) and code review process._

  

Please review: 

### QA Checklist
        
        - [ ] After pulling down and activating this codemod, verify it works
          > Use the command `pub global run over_react_codemod:dart2_9_upgrade --check-for-transitioning`
          - [ ] Run the codemode with the new flag on a project without the new updates. This should return normally with no changes.
          - [ ] Run the codemod without the new flag (to update the repo)
          - [ ] Run the codemod with the new flag again. This should log that it found new instances of the boilerplate, search for changes, and return normally
          - [ ] Manually remove a boilerplate change (so that it's the old boilerplate again)
          - [ ] Run the codemod with the new flag. This should end with a severe log indicating that the process failed

## Merge Checklist
While we perform many automated checks before auto-merging, some manual checks are needed:


[contributing-review-types]: https://github.com/Workiva/over_react_codemod/blob/master/CONTRIBUTING.md#review-types
[contributing-manual-testing]: https://github.com/Workiva/over_react_codemod/blob/master/CONTRIBUTING.md#manual-testing-criteria
